src/Anthropic/Services/Beta/Messages/AnthropicBetaClientExtensions.cs

```diff
@@ -19,6 +20,85 @@ namespace Microsoft.Extensions.AI;
+    /// <remarks>
+    /// <para>
+    /// This method sets the <see cref="ChatOptions.RawRepresentationFactory"/> to include the skills
+    /// in a container parameter. The factory creates a <see cref="MessageCreateParams"/> instance
+    /// with the specified skills and the required beta headers.
+    /// </para>
+    /// <para>
+    /// Skills allow Claude to use specialized capabilities like document creation (pptx, xlsx, docx, pdf)
+    /// or custom user-defined skills. Skills require the code execution tool to be enabled, which will
+    /// be automatically added if not already present in <see cref="ChatOptions.Tools"/>.
+    /// </para>
+    /// <para>
+    /// The following beta headers are automatically included:
+    /// <list type="bullet">
+    /// <item><description><c>code-execution-2025-08-25</c> - Enables code execution (required for Skills)</description></item>
+    /// <item><description><c>skills-2025-10-02</c> - Enables Skills API</description></item>
+    /// </list>
+    /// </para>
+    /// <para>
+    /// Example usage:
+    /// <code>
+    /// var options = new ChatOptions { ModelId = "claude-sonnet-4-5-20250929", MaxOutputTokens = 4096 }
+    ///     .WithSkills([
+    ///         new BetaSkillParams { Type = BetaSkillParamsType.Anthropic, SkillID = "pptx", Version = "latest" }
+    ///     ]);
+    /// </code>
+    /// </para>
+    /// </remarks>
+    /// <exception cref="ArgumentNullException">
+    /// <paramref name="options"/> or <paramref name="skills"/> is <see langword="null"/>.
+    /// </exception>
+    public static ChatOptions WithSkills(
+        this ChatOptions options,
+        IReadOnlyList<BetaSkillParams> skills
+    )
+    {
+        if (options is null)
+        {
+            throw new ArgumentNullException(nameof(options));
+        }
+
+        if (skills is null)
+        {
+            throw new ArgumentNullException(nameof(skills));
+        }
+
+        options.RawRepresentationFactory = _ =>
+            new MessageCreateParams
+            {
+                Model =
+                    options.ModelId
+                    ?? throw new InvalidOperationException(
+                        "Model ID must be specified in ChatOptions when using WithSkills."
+                    ),
+                MaxTokens =
+                    options.MaxOutputTokens
+                    ?? throw new InvalidOperationException(
+                        "MaxOutputTokens must be specified in ChatOptions when using WithSkills."
+                    ),
+                Messages = [],
+                Container = new BetaContainerParams() { Skills = skills },
+                // Required beta headers for skills
+                Betas =
+                [
+                    "code-execution-2025-08-25", // Enables code execution (required for Skills)
+                    AnthropicBeta.Skills2025_10_02, // Enables Skills API

```

Comment:
Separate from this PR, currently the IChatClient implementation doesn't automatically opt-into any of these... should it? i.e. should it try to detect which features are being used and ensure the appropriate feature flags are included? Should it always include all of them?